### PR TITLE
[price-pusher] sui

### PR DESCRIPTION
**Pull request by 0xfirefist**

### CODE REVIEW

File: price_pusher/src/sui/sui.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      });
+
+      if (
+        priceInfoObject.data === undefined ||
+        priceInfoObject.data.content === undefined
+      )
+        throw new Error("Price not found on chain for price id " + priceId);
+
+      if (priceInfoObject.data.content.dataType !== "moveObject")
+        throw new Error("fetched object datatype should be mmoveObject");
```
Comment (jayantk) [inline price_pusher/src/sui/sui.ts:60]: typo: moveObject

File: price_pusher/src/sui/sui.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    // using the hot potato vector.
+    for (const priceId of priceIds) {
+      let priceInfoObjectId;
+      try {
+        priceInfoObjectId = await priceIdToPriceInfoObjectId(
+          this.signer.provider,
+          this.pythPackageId,
+          this.priceFeedToPriceInfoObjectTableId,
+          priceId
+        );
```
Comment (jayantk) [inline price_pusher/src/sui/sui.ts:160]: i'm pretty sure this mapping is static, so you could precompute it once and then use the cached version. That would reduce the number of API calls that this code makes.

**Review by jayantk — APPROVED** [review]
Nicely done. I have a couple minor comments. One other ask would be to document how to run this for sui in the README

File: price_pusher/src/sui/sui.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        priceInfoObject.data.content.fields.price_info.fields.price_feed.fields
+          .price.fields.conf;
+
+      const timestamp =
+        priceInfoObject.data.content.fields.price_info.fields.price_feed.fields
+          .price.fields.timestamp;
+
+      return {
+        price: negative ? "-" + magnitude : magnitude,
+        conf,
```
[the remark this answers is not in the document]
Comment (0xfirefist) [reply price_pusher/src/sui/sui.ts:76]: yes

**Pull request merged**